python-service/tests/conftest.py

```
@@ -0,0 +1,168 @@
+def rustfs_server() -> Iterator[RustFSTestStore]:
+    if shutil.which("docker") is None:
+        pytest.skip("docker is required for RustFS-backed tests")
+
+    api_port = _free_port()
+    console_port = _free_port()
+    endpoint = f"127.0.0.1:{api_port}"
+    access_key = os.environ.get("RUSTFS_ROOT_USER", "rustfsadmin")
+    secret_key = os.environ.get("RUSTFS_ROOT_PASSWORD", "rustfsadmin")
+    container_name = f"anomaly-python-service-rustfs-{api_port}"
+    image = os.environ.get("RUSTFS_DOCKER_IMAGE", "rustfs/rustfs:latest")
+    process = subprocess.Popen(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)